core/src/main/java/io/questdb/griffin/CompiledQueryImpl.java

```diff
@@ -100,12 +101,18 @@ public OperationFuture execute(SqlExecutionContext sqlExecutionContext, SCSequen
         switch (type) {
             case INSERT:
                 return insertOp.execute(sqlExecutionContext);
+            case CREATE_TABLE:
+            case CREATE_TABLE_AS_SELECT:
+                assert false;
```

Comment:
should this be a proper throw? using `assert` means a vastly different behaviour with and without `-ea`

Reply:
if it's a bad API usage then why not to throw an exception unconditionally? `-ea` or not. 
asserts are good for checking invariants. if it's a bad API usage then it should always throw an execption. why not?

Reply:
that assert should never happen :)  we use asserts everywhere, there isn't really something we have to say for the user. I added a "break" so there i no side effect with `-da`

Reply:
removed side-effect here: https://github.com/questdb/questdb/pull/5159/commits/c44db28a716a873faaa4c76d7eaad2c55359aacc